### PR TITLE
Feature/drawer

### PR DESCRIPTION
**Pull request by DannyMoerkerke**
## What I did

1. implemented lion-drawer component

### CODE REVIEW

File: docs/components/drawer/src/demo-dialog-style.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { css } from '@lion/core';
```
Comment (gerjanvangeest) [inline docs/components/drawer/src/demo-dialog-style.js:1]: I believe this file can be deleted

File: docs/components/drawer/use-cases.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
+# Drawer >> Use Cases ||20
+
+`lion-drawer`
```
Comment (gerjanvangeest) [inline docs/components/drawer/use-cases.md:3]: Make this a nice introduction paragraph?

File: packages/drawer/package.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@lion/drawer",
+  "version": "0.0.1",
```
Comment (gerjanvangeest) [inline packages/drawer/package.json:3]: This should stay to 0.0.0

```suggestion
  "version": "0.0.0",
```

File: packages/drawer/src/drawerStyle.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    display: block;
+    height: 100%;
+    --min-width: 72px;
+    --max-width: 320px;
+    --min-height: auto;
+    --max-height: fit-content;
+    --start-width: var(--min-width);
+    --start-height: 100%;
+    --transition-property: width;
+    --background-color: #ffffff;
```
Comment (gerjanvangeest) [inline packages/drawer/src/drawerStyle.js:14]: I think we can remove more styling here:

background-color, border-radius, paddings, margin...

File: packages/drawer/src/LionDrawer.js [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    return html`
+      <div class="container">
+        <div class="headline-container">
+          <slot name="invoker"></slot>
+          <slot name="headline"></slot>
+        </div>
+        <div class="content-container">
+          <slot name="content"></slot>
+        </div>
+        <slot name="bottom-invoker"></slot>
```
Comment (gerjanvangeest) [inline packages/drawer/src/LionDrawer.js:241]: Not sure about the need of 2 invoker slots in lion. It is intended to be extend the component, and normally speaking only 1 invoker is needed. They can override the template to change the order (content first and invoker later).
Makes the code much simpler too.

File: packages/drawer/src/LionDrawer.js [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+      if (this.__contentNode) {
+        this.__contentNode.style.setProperty(prop, '');
+      }
+    });
+  }
+
+  /**
+   * Update aria labels on state change.
+   * @param {String} position
```
Comment (gerjanvangeest) [inline packages/drawer/src/LionDrawer.js:105]: It's unclear which positions are available.
Also if you change the position, not all styling is set correctly. (tested top -> left).

File: docs/components/drawer/src/demo-dialog-style.js (hunk 1 above)
Comment (DannyMoerkerke) [reply docs/components/drawer/src/demo-dialog-style.js]: fixed

File: docs/components/drawer/use-cases.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+      return import('../icon/assets/iconset-space.js').then(module => module[name]);
+    case 'misc':
+      return import('../icon/assets/iconset-misc.js').then(module => module[name]);
+    default:
+      throw new Error(`Unknown iconset ${iconset}`);
+  }
+});
+```
+
+## Default left
```
[the remark this answers is not in the document]
Comment (DannyMoerkerke) [reply docs/components/drawer/use-cases.md]: fixed

File: docs/components/drawer/use-cases.md [hunk 8]
```diff
@@ -0,0 +1,271 @@
+# Drawer >> Use Cases ||20
+
+`lion-drawer`
+
+```js script
+import { html } from '@mdjs/mdjs-preview';
+import { icons } from '@lion/icon';
+import '@lion/drawer/define';
+import '@lion/icon/define';
+import { demoStyle } from './src/demoStyle.js';
```
[the remark this answers is not in the document]
Comment (DannyMoerkerke) [reply docs/components/drawer/use-cases.md]: fixed

File: packages/drawer/package.json (hunk 3 above)
Comment (DannyMoerkerke) [reply packages/drawer/package.json]: fixed

File: packages/drawer/src/LionDrawer.js (hunk 6 above)
Comment (DannyMoerkerke) [reply packages/drawer/src/LionDrawer.js:105]: I checked again with designers and we don't need them at the top and bottom simultaneously so I will change this.

File: packages/drawer/src/LionDrawer.js (hunk 5 above)
Comment (DannyMoerkerke) [reply packages/drawer/src/LionDrawer.js:241]: I checked again with designers and we don't need them at the top and bottom simultaneously so I will remove this and to the implementation for it in ing-web.

File: docs/components/drawer/use-cases.md [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+
+icons.addIconResolver('lion', (iconset, name) => {
+  switch (iconset) {
```
[the remark this answers is not in the document]
Comment (DannyMoerkerke) [reply docs/components/drawer/use-cases.md:10]: fixed

File: packages/drawer/src/LionDrawer.js (hunk 5 above)
Comment (DannyMoerkerke) [reply packages/drawer/src/LionDrawer.js]: fixed

File: packages/drawer/src/LionDrawer.js [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+
+    if (this.position === 'top') {
+      this.minHeight = host ? getComputedStyle(host).getPropertyValue('--min-height') : '0px';
+      this.maxHeight = host ? getComputedStyle(host).getPropertyValue('--max-height') : '0px';
+      this.minWidth = 'auto';
+      this.maxWidth = 'auto';
+    }
+
+    if (this.position === 'left' || this.position === 'right') {
+      this.minWidth = host ? getComputedStyle(host).getPropertyValue('--min-width') : '0px';
```
Comment (gerjanvangeest) [inline packages/drawer/src/LionDrawer.js:76]:
```suggestion
    } else {
      this.minWidth = host ? getComputedStyle(host).getPropertyValue('--min-width') : '0px';
```

File: docs/components/drawer/overview.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+import { html } from '@mdjs/mdjs-preview';
+import { icons } from '@lion/icon';
+import '@lion/drawer/define';
+import '@lion/icon/define';
+import { demoStyle } from './src/demoStyle.js';
+
+icons.addIconResolver('lion', (iconset, name) => {
+  switch (iconset) {
+    case 'bugs':
+      return import('../icon/assets/iconset-bugs.js').then(module => module[name]);
```
Comment (gerjanvangeest) [inline docs/components/drawer/overview.md:19]: Remove the same here as in the use-cases file

File: docs/components/drawer/overview.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+      consectetur a, blandit eu purus. Donec ut gravida libero, sed accumsan arcu.
+    </div>
+  </div>
+`;
+```
+
+## Features
+
+- Show content when clicking the invoker
+- ...
```
Comment (gerjanvangeest) [inline docs/components/drawer/overview.md:63]: Add more features? or remove this line

File: packages/drawer/src/LionDrawer.js (hunk 6 above)
Comment (gerjanvangeest) [reply packages/drawer/src/LionDrawer.js:93]: Why not:

```js
* @param {'top'|'right'|'left'} position
```

**Comment by gerjanvangeest** [comment]
Missing changeset file

**Review by gerjanvangeest — APPROVED** [review]

**Comment by DannyMoerkerke** [comment]
This was moved to https://github.com/ing-bank/lion/pull/1813

**Pull request closed without merging**